docs/manual/java/guide/cluster/UsingAkkaPersistenceTyped.md

```diff
@@ -0,0 +1,185 @@
+
+We use a simplified shopping cart example to guide you through the process. You can find a full-fledged shopping cart sample on our [samples repository](https://github.com/lagom/lagom-samples/tree/1.6.x/shopping-cart/shopping-cart-java).
+
+## Encoding the model
+
+First, create a class (ie: `ShoppingCartEntity`) extending the abstract class `EventSourcedBehaviorWithEnforcedReplies`.  It's recommended to use `EventSourcedBehaviorWithEnforcedReplies` when modeling a CQRS Aggregate. Using [enforced replies](https://doc.akka.io/docs/akka/2.6/typed/persistence.html?language=Java#replies) requires command handlers to return a `ReplyEffect` forcing the developers to be explicit about replies.
+
+@[shopping-cart-entity](code/docs/home/persistence/ShoppingCartEntity.java)
+
+`EventSourcedBehaviorWithEnforcedReplies` has three type parameters (Command, Event and State) and a few mandatory and optional methods that will be implemented as we progress through the guide.
+
+All model classes (`ShoppingCartEntity.Command`, `ShoppingCartEntity.Event`, `ShoppingCartEntity.ShoppingCart`) will be defined as static inner classes of `ShoppingCartEntity`.
+
+### Modelling the State
+
+The state of the shopping cart is defined as following:
+
+@[shopping-cart-state](code/docs/home/persistence/ShoppingCartEntity.java)
+
+The `ShoppingCart` has a `checkedOutTime` that can be set when transitioning from one state (open shopping cart) to another (checked-out shopping cart). As we will see later, each state encodes the commands it can handle, which events it can persist, and to which other states it can transition.
+
+> **Note**: This is not the recommended technique, though. Whenever your model goes through different state transitions, the recommendation is to have a trait and extensions of it for each state. See examples in the [style guide for Akka Persistence Typed](https://doc.akka.io/docs/akka/2.6/typed/persistence-style.html?language=Java).
+
+### Modelling Commands and Replies
+
+Next, we define the commands that we can send to it.
+
+Each command defines a [reply](https://doc.akka.io/docs/akka/2.6/typed/persistence.html?language=Java#replies) through a `ActorRef<R> replyTo` field where `R` is the reply that will be sent back to the caller. Replies are used to communicate back if a command was accepted or rejected or to read the aggregate data (ie: read-only commands). It is also possible to have a mix of both. For example, if the command succeeds, it returns some updated data; if it fails, it returns a rejected message. Or you can have commands without replies (ie: fire-and-forget). This is a less common pattern in CQRS Aggregate modeling though and not covered in this example.
```

Comment:
```suggestion
Each command defines a [reply](https://doc.akka.io/docs/akka/2.6/typed/persistence.html?language=Java#replies) through a `ActorRef<R> replyTo` field where `R` is the reply type that will be sent back to the caller. Replies are used to communicate back if a command was accepted or rejected or to read the aggregate data (ie: read-only commands). It is also possible to have a mix of both. For example, if the command succeeds, it returns some updated data; if it fails, it returns a rejected message. Or you can have commands without replies (ie: `fire-and-forget`). This is a less common pattern in CQRS Aggregate modeling though and not covered in this example.
```

 - `s/where `R` is the reply type that/where `R` is the reply *type* that`
 - backticks in `fire-and-forget`. I wanted this to have the same style to `at-most-once`but we're inconsistent in the format of `at-most-once` vs (plain style) at-most-once. Still, it looks like some styling (`code`, _italics_, ...) should be used. 🤔